repos/system_upgrade/el7toel8/actors/networkmanagerreadconfig/libraries/library.py

```diff
@@ -0,0 +1,55 @@
+import os
+
+from six.moves.configparser import ConfigParser, ParsingError
+
+from leapp.libraries.stdlib import CalledProcessError, run, api
+
+
+def _is_file_readable(path):
+    ''' Verify if file exists and is readable '''
+    return os.path.isfile(path) and os.access(path, os.R_OK)
+
+
+def read_nm_config(file_path=None):
+    # import pdb
+    # pdb.set_trace()
+    # file_path = os.path.join(os.getcwd(), file_path)
+    if file_path:
+        if _is_file_readable(file_path):
+            with open(file_path, 'r') as f:
+                r = f.read()
+                return r
+        else:
+            return None
+    else:
+        try:
+            # Use 'NM --print-config' to read the configurationo so
+            # that the main configuration file and other files in
+            # various directories get merged in the right way.
+            r = run(['NetworkManager', '--print-config'], split=False)['stdout']
+            return r
+        except (OSError, CalledProcessError) as e:
+            api.current_logger().warning('Error reading NetworkManager configuration: {}'.format(e))
+            return None
```

Comment:
If NM config cannot be read the actor execution should stop (producing empty `NetworkManagerConfig` message). However when `read_nm_config` returns `None`, subsequent call to `parser_nm_config` will fail because `StringIO` expects string or buffoer, not `None`.